src/renderer/services/chain/address.ts

```diff
@@ -0,0 +1,47 @@
+import { Chain } from '@thorchain/asgardex-util'
+import * as FP from 'fp-ts/lib/function'
+import * as O from 'fp-ts/lib/Option'
+import * as Rx from 'rxjs'
+import * as RxOp from 'rxjs/operators'
+
+import { BASE_CHAIN } from '../../const'
+import { eqChain } from '../../helpers/fp/eq'
+import * as BNB from '../binance/service'
+import * as BTC from '../bitcoin/context'
+import { selectedPoolChain$ } from '../midgard/common'
+import { AddressRx } from './types'
+
+const addressByChain$ = (chain: Chain): AddressRx => {
+  switch (chain) {
+    case 'BNB':
+      return BNB.address$
+    case 'BTC':
+      return BTC.address$
+    case 'ETH':
+      return Rx.of(O.none)
+    case 'THOR':
+      return Rx.of(O.none)
+  }
+}
+
+/**
+ * Users wallet address for cross-chain
+ */
+const crossAddress$: AddressRx = selectedPoolChain$.pipe(
+  RxOp.switchMap((oChain) =>
+    FP.pipe(
+      oChain,
+      // x-chain only
+      O.filter((chain) => !eqChain.equals(chain, BASE_CHAIN)),
+      O.fold(() => Rx.of(O.none), addressByChain$)
+    )
+  ),
+  RxOp.distinctUntilChanged()
```

Comment:
as an inner value is an ADT (in this case Option) such usage of `distinctUntilChanged` will affect nothing as there are always new pointers. this should be
```ts
RxOp.distinctUntilChanged(eqOString.equals)
```